src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -109,12 +140,21 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
 }
 
 object Iterator {
+
   val empty: Iterator[Nothing] = new Iterator[Nothing] {
     def hasNext = false
     def next() = throw new NoSuchElementException("next on empty iterator")
   }
+
+  def single[A](a: A): Iterator[A] = new Iterator[A] {
+    private var consumed: Boolean = false
+    def hasNext = !consumed
+    def next() = if (consumed) empty.next() else { consumed = true; a }
+  }
+
   def apply[A](xs: A*): Iterator[A] = new IndexedView[A] {
```

Comment:
The `IndexedView` is created just so you can get an `iterator` on it.  So you take elements, create an object to hold them, then create _another_ object to point to the first object, with all sorts of indirection on the way.  `IndexedView` is just a box around `xs` providing `iterator`, but it's so simple to write again that we should rather than doubling all allocations, indirections, etc..